libraries/WebServer/src/Parsing.cpp

```diff
@@ -186,8 +186,12 @@ bool WebServer::_parseRequest(NetworkClient &client) {
       _currentHandler->raw(*this, _currentUri, *_currentRaw);
       _currentRaw->status = RAW_WRITE;
 
-      while (_currentRaw->totalSize < _clientContentLength) {
-        _currentRaw->currentSize = client.readBytes(_currentRaw->buf, HTTP_RAW_BUFLEN);
+      while (1) {
+        size_t read_len = std::min(_clientContentLength - _currentRaw->totalSize, (size_t) HTTP_RAW_BUFLEN);
+        if (read_len == 0) {
+          break;
+        }
+        _currentRaw->currentSize = client.readBytes(_currentRaw->buf, read_len);
```

Comment:
```suggestion
      while (_currentRaw->totalSize < _clientContentLength) {
        size_t read_len = std::min(_clientContentLength - _currentRaw->totalSize, (size_t) HTTP_RAW_BUFLEN);
        _currentRaw->currentSize = client.readBytes(_currentRaw->buf, read_len);
```

Reply:
It's a microoptimization - it's faster to compare against zero.